### PR TITLE
Primary caching 6: TextLogs & TimeSeries

### PR DESCRIPTION
**Pull request by teh-cmc**
Integrates the cached APIs with the TextLog & TimeSeries views, which is pretty trivial. 

This of course does nothing, since the cache doesn't cache range queries yet.

---

Part of the primary caching series of PR (index search, joins, deserialization):
- #4592
- #4593
- #4659
- #4680 
- #4681
- #4698
- #4711
- #4712
- #4721 
- #4726 

---

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4592/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4592/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4592/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4592)
- [Docs preview](https://rerun.io/preview/f23f605d1e78932870ac58e9c806ca13957839dd/docs) 
- [Examples preview](https://rerun.io/preview/f23f605d1e78932870ac58e9c806ca13957839dd/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_viewer_context/src/app_options.rs [hunk 1]
```diff
@@ -61,6 +64,10 @@ impl Default for AppOptions {
             // merged in.
             experimental_primary_caching_point_clouds: false,
 
+            // TODO(cmc): default to true for debug/rerun-workspace once minimal features have been
+            // merged in.
```
Comment (teh-cmc) [inline crates/re_viewer_context/src/app_options.rs:68]: Will be defaulted to `true` later this week when cached range support starts becoming a thing.

File: crates/re_types/definitions/rerun/archetypes/text_log.fbs [hunk 2]
```diff
@@ -22,5 +22,5 @@ table TextLog (
   level: rerun.components.TextLogLevel ("attr.rerun.component_recommended", nullable, order: 200);
 
   /// Optional color to use for the log line in the Rerun Viewer.
-  color: rerun.components.Color (nullable, order: 300);
+  color: rerun.components.Color ("attr.rerun.component_optional", nullable, order: 300);
```
Comment (teh-cmc) [inline crates/re_types/definitions/rerun/archetypes/text_log.fbs:25]: Should probably open an issue for this

File: crates/re_space_view_text_log/src/visualizer_system.rs [hunk 3]
```diff
@@ -66,28 +65,27 @@ impl VisualizerSystem for TextLogSystem {
...
-                }
-            }
+                |((time, row_id), _, bodies, levels, colors)| {
+                    for (body, level, color) in
+                        itertools::izip!(bodies.iter(), levels.iter(), colors.iter())
+                    {
+                        self.entries.push(Entry {
+                            row_id,
+                            entity_path: data_result.entity_path.clone(),
+                            // TODO(cmc): real support for timeless data in caches.
```
Comment (teh-cmc) [inline crates/re_space_view_text_log/src/visualizer_system.rs:80]: - Implemented in https://github.com/rerun-io/rerun/pull/4721

File: crates/re_types/definitions/rerun/archetypes/text_log.fbs (hunk 2 above)
Comment (teh-cmc) [reply crates/re_types/definitions/rerun/archetypes/text_log.fbs:25]: - #4739

**Review by Wumpf — APPROVED** [review]
after the point one quite trivial indeed 😄

**Pull request merged**